### ISSUE TITLE
Make GenC examples go through for verification

### ISSUE THREAD

**Issue by jad-hamza** [epfl-lara/stainless#926]
The examples in https://github.com/epfl-lara/stainless/tree/master/frontends/benchmarks/genc do not verify properly with Stainless, we should fix them and add them to a verification test suite.

**Comment by vkuncak** [comment]
@mario-bucev you made progress on this, right?

**Comment by mario-bucev** [comment]
Yes, though, as we have discussed, we should move the LZWa benchmark to bolts (LZWb is similar to LZWa, but I haven't looked at it yet).
For that, at least the following issues should be addressed:
- #1272
- #1270
- #1269
- #1268
I already have the fixes and will create a PR once #1265 is merged.
